src/site/content/en/mini-apps/mini-app-example-project/index.md

```diff
@@ -0,0 +1,256 @@
+
+```js
+import Page from '../page.js';
+
+const page = new Page({
+  eventHandlers: {
+    back: (e) => {
+      e.preventDefault();
+      window.top.history.back();
+    },
+  },
+});
+```
+
+<figure class="w-figure">
+  <img src="./iframe.png" width="500" alt="In-app page realized as an iframe.">
+  <figcaption class="w-figcaption">
+    Navigation happens from iframe to iframe.
+  </figcaption>
+</figure>
+
+## Styling
+
+Styling of pages happens per-page in its own scoped CSS file.
+This means elements can usually just be directly addressed by their element names,
+since no clashes with other pages can occur.
+Global styles are added to each page, so central settings like the `font-family` or the `box-sizing`
+do not need to be declared repeatedly.
+This is also where the themes and dark mode options are defined.
+The listing below shows the rules for the Preferences page that lays out the various form elements
+on a grid.
+
+```css
+main {
+  max-width: 600px;
+}
+
+form {
+  display: grid;
+  grid-template-columns: auto 1fr;
+  grid-gap: 0.5rem;
+  margin-block-end: 1rem;
+}
+
+label {
+  text-align: end;
+  grid-column: 1 / 2;
+}
+
+input,
+select {
+  grid-column: 2 / 3;
+}
+```
+
+<figure class="w-figure">
+  <img src="./preferences.png" width="500" alt="HIIT Time app preferences page showing a form in grid layout.">
+  <figcaption class="w-figcaption">
+    Every page is its own world. Styling happens directly with the element names.
+  </figcaption>
+</figure>
+
+## Screen wake lock
+
+During a workout, the screen should not turn off.
+On browsers that support it, HIIT Time realizes this through a [screen wake lock](/wake-lock/).
+The snippet below shows how it is done.
+
+```js
+if ('wakeLock' in navigator) {
+  const requestWakeLock = async () => {
+    try {
+      page.shared.wakeLock = await navigator.wakeLock.request('screen');
+      page.shared.wakeLock.addEventListener('release', () => {
+        // Nothing.
+      });
+    } catch (err) {
+      console.error(`${err.name}, ${err.message}`);
+    }
+  };
+  // Request a screen wake lock…
+  await requestWakeLock();
+  // …and re-request it when the page becomes visible.
+  document.addEventListener('visibilitychange', async () => {
+    if (
+      page.shared.wakeLock !== null &&
+      document.visibilityState === 'visible'
+    ) {
+      await requestWakeLock();
+    }
+  });
+}
+```
+
+## Testing the application
+
+The HIIT Time application is available on [GitHub](https://github.com/tomayac/hiit-time).
+You can play with the [demo](https://tomayac.github.io/hiit-time/) in a new window,
+or right in the iframe embed below, which simulates a mobile device.
+
+<iframe src="https://tomayac.github.io/hiit-time/#workout" width="411" height="731" loading="lazy" frameborder="0" allow="screen-wake-lock"></iframe>
+
+{% Banner 'neutral' %}
+  👉 The final chapter ends this collection on mini apps with a [conclusion](mini-app-conclusion).
```

Comment:
404

Reply:
Ooops, this was missing a leading slash.